src/Data/CAS/RocksDB.hs

```diff
@@ -757,3 +770,79 @@ decIterKey it k = case B.splitAt (B.length namespace) k of
     namespace = _rocksDbTableIterNamespace it
 {-# INLINE decIterKey #-}
 
+data Checkpoint
+
+foreign import ccall unsafe "rocksdb\\c.h rocksdb_checkpoint_object_create" 
+  rocksdb_checkpoint_object_create :: C.RocksDBPtr -> Ptr CString -> IO (Ptr Checkpoint)
+
+foreign import ccall unsafe "rocksdb\\c.h rocksdb_checkpoint_create"
+  rocksdb_checkpoint_create :: Ptr Checkpoint -> CString -> CULong -> Ptr CString -> IO ()
+
+foreign import ccall unsafe "rocksdb\\c.h rocksdb_checkpoint_object_destroy"
+  rocksdb_checkpoint_object_destroy :: Ptr Checkpoint -> IO ()
+
+checked :: String -> Ptr CString -> IO a -> IO a
+checked whatWasIDoing errPtr act = do
+  r <- act
+  err <- peek errPtr
+  unless (err == nullPtr) $ do
+    errStr <- B.packCString err
+    let msg = unwords ["error while", whatWasIDoing <> ":", B8.unpack errStr]
+    free err
+    fail msg
```

Comment:
I think, this translates into `error`? In that case you may add a `HasCallStack` constraint to the function call for better stack tracing.